homeassistant/components/insteon/__init__.py

```diff
@@ -468,22 +468,17 @@ def group(self):
     @property
     def unique_id(self) -> str:
         """Return a unique ID."""
-        if self._insteon_device_state.group == 0x01:
-            uid = self._insteon_device.id
-        else:
-            uid = '{:s}_{:d}'.format(self._insteon_device.id,
-                                     self._insteon_device_state.group)
-        return uid
+        return self._generate_network_address()
 
     @property
     def name(self):
         """Return the name of the node (used for Entity_ID)."""
-        name = ''
-        if self._insteon_device_state.group == 0x01:
-            name = self._insteon_device.id
+        addr = self._generate_network_address()
+        if self._insteon_device.description is None:
+            name = addr
         else:
-            name = '{:s}_{:d}'.format(self._insteon_device.id,
-                                      self._insteon_device_state.group)
+            name = '{:s} {:s}'.format(self._insteon_device.description,
+                                      addr)
```

Comment:
Is `name` expected to be a human readable name? If so, can I suggest using the following:
```
name = '{:s} {:s} Group {:d}'.format(self._insteon_device.description,
                                     self._insteon_device.address.human,
                                     self._insteon_device.group)
```
This will create a `name` that looks like `KeyPadLinc Dimmer 1A.2B.3C Group 5`

Reply:
The Human address is great and will add it in.  As well I'm looking at making it even more contextual, adding "Fan"/"Light" for a FanLinc or "Relay"/"Sensor" for an I/OLinc  instead of the group number id.  Ill post an update to the PR soon.